routers/web/repo/actions/view.go

```diff
@@ -304,6 +306,42 @@ func ViewPost(ctx *context_module.Context) {
 	ctx.JSON(http.StatusOK, resp)
 }
 
+// getWorkflowFileLink return url for the source of the workflow file that was run
+func getWorkflowFileLink(ctx *context_module.Context, run *actions_model.ActionRun) string {
```

Comment:
Shouldn't this method belong to `ActionRun`?

Reply:
Yeah, I think it rather belongs into `models`.
After all, there might be other places where we want to have this information.

Reply:
@william-allspice tag me when that's all set (or if there is any issue) and I can review.

Reply:
@delvh I was able to get this refactored into the actions module (not model). This will allow it to be reused.

The logic in this function depends on other functions in the actions module (mainly ListWorkflows). I believe that logic lives in modules because today there is no database table representing a workflow, this is a similar caveat to what was discussed earlier in this thread regarding an ActionRun.WorkflowID really just representing the name of the workflow file at the time the action was run.

Like this, all of the workflow file logic lives in the same place, the action module. Do you agree this is acceptable @delvh ?

Reply:
@kdumontnu, changes implemented.